website/project/model.py

```diff
@@ -635,6 +635,11 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
 
     creator = fields.ForeignField('user', backref='created')
     contributors = fields.ForeignField('user', list=True, backref='contributed')
+    # Time of last sent notification email to newly added contributors
+    # {<User._id>:
+    #   {'last_sent': time.time()}
+    # }
+    contributor_record = fields.DictionaryField(default=dict)
```

Comment:
Both implementations seemed ok, so I went with the `Node` model. But I think adding the field to the `User` model would solve some issues I ran into with saving the node inside of `notify_contributor` -- I'll make those changes.